python/ray/data/dataset.py

```diff
@@ -602,6 +602,47 @@ def drop_columns(
             lambda batch: batch.drop(columns=cols), compute=compute, **ray_remote_args
         )
 
+    def select_columns(
+        self,
+        columns: List[str],
+        *,
+        compute: Union[str, ComputeStrategy] = None,
+        **ray_remote_args,
+    ) -> "Dataset[T]":
+        """Select one or more columns from the dataset.
+
+        Columns passed in will be de-duped since ArrowBlock and PandasBlock
+        `select` does not explicitly handle duplicated columns.
```

Comment:
Can we mention that the columns passed in have to be in the schema?

Reply:
ack. will add - i will also add a schema validation check prior to map_batches to guard against this